### PR TITLE
correct usage of with_vulkan flag in qt/5.15.2

### PR DESCRIPTION
**Pull request by AndreyMlashkin**
Specify library name and version:  **qt/5.15.2**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

fixes conan-io/conan-center-index#6773
---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by AndreyMlashkin** [comment]
It works just fine for me on Linux and Windows, but I've got an error on Mac:

 -f libqsqlite.dylib /Users/rr-admin/.conan/data/qt/5.15.2/andrei/test/build/25d4c97d065f7fc9413fa65c77ac475e8bf34585/build_folder/qtbase/plugins/sqldrivers/libqsqlite.dylib
compiling /Users/rr-admin/.conan/data/qt/5.15.2/andrei/test/source/qt5/qtbase/src/plugins/platforms/cocoa/qcocoanativeinterface.mm
compiling /Users/rr-admin/.conan/data/qt/5.15.2/andrei/test/source/qt5/qtbase/src/plugins/platforms/cocoa/qcocoaeventdispatcher.mm
In file included from /Users/rr-admin/.conan/data/qt/5.15.2/andrei/test/source/qt5/qtbase/src/plugins/platforms/cocoa/qcocoawindowmanager.mm:41In file included from /Users/rr-admin/.conan/data/qt/5.15.2/andrei/test/source/qt5/qtbase/src/plugins/platforms/cocoa/qnsview.mm:43:
/Users/rr-admin/.conan/data/qt/5.15.2/andrei/test/source/qt5/qtbase/src/plugins/platforms/cocoa/qcocoawindow.h:56:10: fatal error: 'MoltenVK/mvk_vulkan.h' file not found
#include <MoltenVK/mvk_vulkan.h>
         ^~~~~~~~~~~~~~~~~~~~~~~
:
/Users/rr-admin/.conan/data/qt/5.15.2/andrei/test/source/qt5/qtbase/src/plugins/platforms/cocoa/qcocoawindow.h:56:10: fatal error: 'MoltenVK/mvk_vulkan.h' file not found
#include <MoltenVK/mvk_vulkan.h>
         ^~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
make[5]: *** [.obj/qcocoawindowmanager.o] Error 1
make[5]: *** Waiting for unfinished jobs....
In file included from /Users/rr-admin/.conan/data/qt/5.15.2/andrei/test/source/qt5/qtbase/src/plugins/platforms/cocoa/qnswindow.mm:43:
/Users/rr-admin/.conan/data/qt/5.15.2/andrei/test/source/qt5/qtbase/src/plugins/platforms/cocoa/qcocoawindow.h:56:10: fatal error: 'MoltenVK/mvk_vulkan.h' file not found
#include <MoltenVK/mvk_vulkan.h>
         ^~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
make[5]: *** [.obj/qnswindow.o] Error 1
In file included from /Users/rr-admin/.conan/data/qt/5.15.2/andrei/test/source/qt5/qtbase/src/plugins/platforms/cocoa/qcocoanativeinterface.mm:41:
/Users/rr-admin/.conan/data/qt/5.15.2/andrei/test/source/qt5/qtbase/src/plugins/platforms/cocoa/qcocoawindow.h:56:10: fatal error: 'MoltenVK/mvk_vulkan.h' file not found
#include <MoltenVK/mvk_vulkan.h>
         ^~~~~~~~~~~~~~~~~~~~~~~
In file included from /Users/rr-admin/.conan/data/qt/5.15.2/andrei/test/source/qt5/qtbase/src/plugins/platforms/cocoa/qnswindowdelegate.mm:42:
/Users/rr-admin/.conan/data/qt/5.15.2/andrei/test/source/qt5/qtbase/src/plugins/platforms/cocoa/qcocoawindow.h:56:10: fatal error: 'MoltenVK/mvk_vulkan.h' file not found
#include <MoltenVK/mvk_vulkan.h>
         ^~~~~~~~~~~~~~~~~~~~~~~
1 error generated.

What should I add?
I don't see, that qt6 recipe add MoltenVK

File: recipes/qt/5.x.x/conanfile.py [hunk 1]
```diff
@@ -507,9 +507,7 @@ def build(self):
             args += ["-opengl dynamic"]
 
         if self.options.get_safe("with_vulkan", False):
-            args.append("-vulkan")
-        else:
```
Comment (ericLemanissier) [inline recipes/qt/5.x.x/conanfile.py:511]: We have to keep the else to disable explicitly vulkan, even in cases where qt detects system vulkan

File: recipes/qt/5.x.x/conanfile.py [hunk 2]
```diff
@@ -507,9 +507,7 @@ def build(self):
             args += ["-opengl dynamic"]
 
         if self.options.get_safe("with_vulkan", False):
-            args.append("-vulkan")
```
Comment (ericLemanissier) [inline recipes/qt/5.x.x/conanfile.py:510]: I'm surprised that the wrong argument was used. Did it produce an error message?

Comment (AndreyMlashkin) [reply recipes/qt/5.x.x/conanfile.py:510]: No.
But could you please try on your own, please?

I would not merge this PR until it works for MacOs.

Comment (ericLemanissier) [reply recipes/qt/5.x.x/conanfile.py:510]: Then you should probably change this PR to draft. I will try to test this on Friday, or it will be next week

Comment (ericLemanissier) [reply recipes/qt/5.x.x/conanfile.py:510]: @AndreyMlashkin please revert this change. the existing argument is correct. `-feature-vulkan` works too, but changing this line it makes git blame harder to use.

**Comment by ericLemanissier** [comment]
can you please add `fixes conan-io/conan-center-index#6773` in the PR description ?

**Comment by AndreyMlashkin** [comment]
in the end, it will be a backport of https://github.com/conan-io/conan-center-index/pull/6835/

File: recipes/qt/5.x.x/conanfile.py [hunk 3]
```diff
@@ -507,7 +507,7 @@ def build(self):
...
+            args.append("-feature-vulkan")
```
Comment (ericLemanissier) [inline recipes/qt/5.x.x/conanfile.py:510]:
```suggestion
            args.append("-vulkan")
```

**Review by ericLemanissier — COMMENTED** [review]
is the moltenvk thing needed here too ?

**Comment by AndreyMlashkin** [comment]
Yes, exactly :)

**Review by prince-chrismc — DISMISSED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**